src/application-navigation/ApplicationNavigation.jsx

```diff
@@ -0,0 +1,192 @@
+import React, {
+  useRef, useCallback, Suspense,
+} from 'react';
+import PropTypes from 'prop-types';
+import TerraApplicationNavigation from 'terra-application-navigation';
+import {
+  titleConfigPropType, navigationItemsPropType, extensionItemsPropType, utilityItemsPropType, userConfigPropType,
+} from 'terra-application-navigation/lib/utils/propTypes';
+
+import ApplicationErrorBoundary from '../application-error-boundary';
+import ApplicationLoadingOverlay, { ApplicationLoadingOverlayProvider } from '../application-loading-overlay';
+import { NavigationPromptCheckpoint, navigationPromptResolutionOptionsShape } from '../navigation-prompt';
+
+const propTypes = {
+  /**
+   * A configuration object that defines the strings rendered within the ApplicationNavigation header.
+   */
+  titleConfig: titleConfigPropType,
+  /**
+   * A configuration object with information specifying the creation of the Extension buttons rendered within the
+   * ApplicationNavigation header.
+   */
+  extensionItems: extensionItemsPropType,
+  /**
+   * A function to be executed upon the selection of an extensions item.
+   * Ex: `onSelectExtensionsItem(String selectedUtilityItemKey, Object metaData)`
+   */
+  onSelectExtensionItem: PropTypes.func,
+  /**
+   * A configuration object with information pertaining to the application's user.
+   */
+  userConfig: userConfigPropType,
+  /**
+   * An element to render within the ApplicationNavigation menu, shifted to the drawer at the `medium` and below.
+   */
+  hero: PropTypes.element,
```

Comment:
What type of element should be supplied to hero? Should this be an image or a control of some sort or an external link?

Reply:
Good question. This is one of those things that is still a little hazy so we left the API open. From the examples I've seen, it's typically an image of some kind, but we don't know enough to make things too strict here.